preview/MsixCore/msixmgr/Package.cpp

```diff
@@ -148,30 +155,120 @@ HRESULT MsixCoreLib::PackageBase::ProcessPSFIfNecessary()
+    std::wstring jsonConfigFile = GetPackageDirectoryPath() + psfConfigFile;
+    bool configFileExists = false;
+    RETURN_IF_FAILED(FileExists(jsonConfigFile, configFileExists));
+
+    if (configFileExists)
     {
-        if (std::experimental::filesystem::is_regular_file(p.path()) && CaseInsensitiveEquals(p.path().extension(), L".json"))
+        // parse the file and see if it has info we need.
+        std::ifstream ifs(jsonConfigFile);
+        IStreamWrapper isw(ifs);
+
+        Document doc;
+        doc.ParseStream(isw);
+
+        if (doc.HasParseError())
         {
-            // parse the file and see if it has info we need.
-            // TODO: fill this in with actual stuff. Hardcoding for this test package for now
-            std::wstring psfExecutable = L"Nokia Siemens Networks\\Managers\\BTS Site\\BTS Site Manager\\jre\\1_6_0\\bin\\javaw.exe";
-            std::wstring psfWorkingDirectory = L"Nokia Siemens Networks\\Managers\\BTS Site\\BTS Site Manager";
-            m_executionInfo.commandLineArguments = L"-splash:Splash_Wn_BTS_Site_Manager.png -cp cl\\cl.jar -client com.nokia.em.poseidon.PoseidonStarter -confFile cl\\CLConf.xml";
-
-            // resolve the given paths from json into full paths.
-            m_executionInfo.resolvedExecutableFilePath = FilePathMappings::GetInstance().GetExecutablePath(psfExecutable, m_packageFullName.c_str());
-            m_executionInfo.workingDirectory = FilePathMappings::GetInstance().GetExecutablePath(psfWorkingDirectory, m_packageFullName.c_str());
-
-            TraceLoggingWrite(g_MsixTraceLoggingProvider,
-                "PSF redirection",
-                TraceLoggingWideString(m_executionInfo.resolvedExecutableFilePath.c_str(), "Resolved PSF executable"),
-                TraceLoggingWideString(m_executionInfo.workingDirectory.c_str(), "WorkingDirectory"),
-                TraceLoggingWideString(m_executionInfo.commandLineArguments.c_str(), "Arguments"));
+            return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
         }
-    }
+        if (doc.HasMember("applications"))
+        {
+            const Value& apps = doc["applications"];
+
+            for (Value::ConstValueIterator itr = apps.Begin(); itr != apps.End(); ++itr)
+            {
+                std::string id = (*itr)["id"].GetString();
+                std::wstring jsonApplicationId;
+                jsonApplicationId.assign(id.begin(), id.end());
+
+                //ApplicationId in json should match applicationId from the xml manifest. Right now, we only process first application from the xml manifest anyway
+                if (CaseInsensitiveEquals(jsonApplicationId, m_applicationId))
+                {
+                    // Fail if endScript is present in current implementation as we do not support those cases yet
+                    if ((*itr).HasMember("endScript"))
+                    {
+                        return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);

```

Comment:
>return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED) [](start = 24, length = 46)

for each case where we return this, add a TraceLoggingWrite() that explains the reason so that somebody looking at logs can tell which scenario we don't support

Reply:
Added tracelogging for runonce = false and presence of endScript.